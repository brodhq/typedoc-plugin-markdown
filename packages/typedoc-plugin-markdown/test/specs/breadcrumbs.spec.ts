import { BreadcrumbsComponent } from '../../src/renderer/components/page/breadcrumbs';
import { TestApp } from '../test-app';

describe(`Breadcrumbs:`, () => {
  let testApp: TestApp;

  beforeAll(async () => {
    testApp = new TestApp(['reflections.ts', 'declarations.ts']);
    await testApp.bootstrap();
  });

  test(`should compile breadcrumbs for a entrypoint'`, () => {
    const page = testApp.getPageByUrl('modules.md').model;
    expect(BreadcrumbsComponent(page)).toMatchSnapshot();
  });

  test(`should compile breadcrumbs for a module'`, () => {
    const page = testApp.getPageByModelName('reflections').model;
    expect(BreadcrumbsComponent(page)).toMatchSnapshot();
  });

  test(`should compile breadcrumbs for a reflection'`, () => {
    const page = testApp.getPageByModelName('ReflectionClass').model;
    expect(BreadcrumbsComponent(page)).toMatchSnapshot();
  });
});
