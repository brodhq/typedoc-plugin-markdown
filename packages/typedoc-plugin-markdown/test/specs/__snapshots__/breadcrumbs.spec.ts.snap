// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs: should compile breadcrumbs for a entrypoint' 1`] = `"Modules"`;

exports[`Breadcrumbs: should compile breadcrumbs for a module' 1`] = `"[Modules](../modules.md) / reflections"`;

exports[`Breadcrumbs: should compile breadcrumbs for a reflection' 1`] = `"[Modules](../modules.md) / [reflections](../modules/reflections.md) / ReflectionClass"`;
